### PR TITLE
Fix Builder API Default Property (#994)

### LINKED ISSUE #994: BUG- Selector not working properly with the Python API
**Description**

If the user attempts to create a selector without specifying the *value* keyword for the bound variable, they will not be able to use the selector. It is mandatory to explicitly declare `value="{selected_value}"`. Contrary to expectations, the initial parameter that describes the visual element is not *value*. Instead, the utilized parameter is the filter property.

image

**How to reproduce**
```python
from taipy.gui import Gui
import taipy.gui.builder as tgb

selected_value = None
lov = ["a", "b", "c"]

with tgb.Page() as page:
    # Not working, I am in fact defining the filter value here
    tgb.selector("{selected_value}", lov=lov)

    # Working
    tgb.selector(value="{selected_value}", lov=lov)


Gui(page).run()
```

image

**Expected behavior**
The first example should work as described in the documentation.

**Runtime environment**
 - Taipy version: 3.1

**Acceptance Criteria**
- [ ] Ensure new code is unit tested, and check code coverage is at least 90%
- [ ] Create related issue in taipy-doc for documentation and Release Notes if relevant

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Resolve #994

### CODE REVIEW

**Review by FabienLelaquais — COMMENTED** [review]
👌👌

**Review by FabienLelaquais — DISMISSED** [review]
👍

**Comment by FabienLelaquais** [comment]
Note for @FredLL-Avaiga 
The important thing is in taipy/gui/builder/_api_generator.py, line 53.

File: taipy/gui/builder/_api_generator.py [hunk 1]
```diff
@@ -46,17 +47,18 @@ def add_default(self):
         module_name = current_frame.f_back.f_globals["__name__"]
         self.__module = module = sys.modules[module_name]
         with open(os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "viselements.json"))) as viselements:
-            data = json.load(viselements)
-            if "blocks" not in data or "controls" not in data:
+            viselements = json.load(viselements)
```
Comment (FredLL-Avaiga) [inline taipy/gui/builder/_api_generator.py]: Is it not a bit confusing to re-use viselements?

File: taipy/gui/builder/_api_generator.py [hunk 2]
```diff
@@ -46,17 +47,18 @@ def add_default(self):
...
+            if "blocks" not in viselements or "controls" not in viselements:
                 raise RuntimeError(f"{error_message}: Invalid viselements.json file.")
-            for blockElement in data["blocks"]:
+            viselements = resolve_inherits(t.cast(VisElements, viselements))
```
Comment (FredLL-Avaiga) [inline taipy/gui/builder/_api_generator.py]: Again?
But I like the cast

**Review by FredLL-Avaiga — DISMISSED** [review]
Please find good names and don't re-use
I like the typing of the viselements
Well done

File: taipy/gui/builder/_api_generator.py (hunk 1 above)
Comment (dinhlongviolin1) [reply taipy/gui/builder/_api_generator.py]: will do! tx Fred

**Review by FredLL-Avaiga — APPROVED** [review]
tx for the changes: way clearer :-)

**Comment by dinhlongviolin1** [comment]
> tx for the changes: way clearer :-)

Nothing makes me happier than you being happy 😄

**Pull request merged**